lib/extensions/ar_migration.rb

```diff
@@ -1,48 +1,35 @@
-module ArPglogicalMigration
-  module PglogicalMigrationHelper
-    def self.migrations_column_present?
-      ActiveRecord::Base.connection.columns("miq_regions").any? { |c| c.name == "migrations_ran" }
-    end
-
-    def self.my_region_number
-      # Use ApplicationRecord here because we need to query region information
-      @my_region_number ||= ApplicationRecord.my_region_number
-    end
-
-    def self.my_region_created?
-      ActiveRecord::Base.connection.exec_query(<<~SQL).first["exists"]
-        SELECT EXISTS(
-          SELECT id FROM miq_regions WHERE region = #{ActiveRecord::Base.connection.quote(my_region_number)}
-        )
-      SQL
+module ArPglogicalMigrationHelper
+  def self.discover_schema_migrations_ran_class
+    return unless ActiveRecord::Base.connection.table_exists?("schema_migrations_ran")
+    Class.new(ActiveRecord::Base) do
+      require 'active_record-id_regions'
+      include ActiveRecord::IdRegions
+      self.table_name = "schema_migrations_ran"
     end
+  end
 
-    def self.update_local_migrations_ran(version, direction)
-      return unless migrations_column_present?
-      return unless my_region_created?
+  def self.update_local_migrations_ran(version, direction)
+    return unless schema_migrations_ran_class = discover_schema_migrations_ran_class
 
-      new_migrations = ActiveRecord::SchemaMigration.normalized_versions
-      new_migrations << version if direction == :up
-      migrations_value = ActiveRecord::Base.connection.quote(PG::TextEncoder::Array.new.encode(new_migrations))
+    new_migrations = ActiveRecord::SchemaMigration.normalized_versions
+    new_migrations << version if direction == :up
 
-      ActiveRecord::Base.connection.exec_query(<<~SQL)
-        UPDATE miq_regions
-        SET migrations_ran = #{migrations_value}
-        WHERE region = #{ActiveRecord::Base.connection.quote(my_region_number)}
-      SQL
+    (new_migrations - schema_migrations_ran_class.pluck(:version)).each do |v|
```

Comment:
This should probably be region aware, so `schema_migrations_ran_class.in_my_region.pluck(:version)`, right?